File: administrative-module/management-administrative-ws/src/main/java/pe/edu/ceid/simi/management/application/progcurso/ProgCursoService.java
package pe.edu.ceid.simi.management.application.progcurso;

import java.util.List;

import pe.edu.ceid.simi.management.domain.progcurso.model.ProgCurso;
import pe.edu.ceid.simi.management.domain.progcurso.model.ProgCursoDTO;

public interface ProgCursoService {
	ProgCurso crearProgCurso(ProgCurso progCurso);
	ProgCurso editProgCurso(ProgCurso progCurso, int id);
	boolean deleteProgCurso(int id);
	List<ProgCursoDTO> getProgCursos();
	ProgCursoDTO getProgCursoById(int id);
}

File: administrative-module/administrative-web/src/app/services/periodo-academico/progCurso.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { ProgCursoDTO } from 'src/app/domain/ProgCursoDTO';
import { ProgCurso } from 'src/app/domain/ProgCurso';

@Injectable({
  providedIn: 'root'
})
export class ProgCursoService {

  private url: string;
  constructor(private http: HttpClient) {
    this.url = 'simi/api/v1/programacionCurso/progcurso';
  }

  getProgCurso() {
    return this.http.get<ProgCursoDTO[]>(this.url + '/' + 'listProgCursos');
  }

  public getProgCursoById(id: number) {
    return this.http.get<ProgCursoDTO>(this.url + '/' + 'obtenerProgCursoById' + '/' + id);
  }
  public getProgCursosPeriodo(idPeriodo: number) {
    return this.http.get<ProgCursoDTO[]>(this.url + '/' + 'listProgCursosByPeriodo' + '/' + idPeriodo);
  }

  public getProgCursosBySedePeriodo(idSede: number,idPeriodo: number) {
    return this.http.get<ProgCursoDTO[]>(this.url + '/' + 'listProgCursosBySedePeriodo' + '/' + idSede+ '/' + idPeriodo);
  }

   public crearProgCurso(pa: ProgCurso) {
     return this.http.post<ProgCurso>(this.url + '/' + 'crearProgCurso', pa);
   }


   public editarProgCursoById(pa: ProgCurso,id: number) {
     return this.http.put<ProgCurso>(this.url + '/' + 'actualizarProgCurso'+ '/' + id, pa);
   }

   public deleteProgCursoById(id: number) {
     return this.http.delete<boolean>(this.url + '/' + 'eliminarProgCursoById' + '/' + id);
   }

}
